go-controller/pkg/ovn/policy.go

```diff
@@ -1148,276 +1159,298 @@ func hasAnyLabelSelector(peers []knet.NetworkPolicyPeer) bool {
 	return false
 }
 
-// createNetworkPolicy creates a network policy
-func (oc *Controller) createNetworkPolicy(np *networkPolicy, policy *knet.NetworkPolicy, aclLogging *ACLLoggingLevels) error {
+type policyHandler struct {
+	gress             *gressPolicy
+	namespaceSelector *metav1.LabelSelector
+	podSelector       *metav1.LabelSelector
+}
 
-	np.Lock()
+// createNetworkPolicy creates a network policy, should be retriable
+// if network policy with given key exists, it will try to clean it up first, and return an error if it fails
+func (oc *Controller) createNetworkPolicy(policy *knet.NetworkPolicy, aclLogging *ACLLoggingLevels) (*networkPolicy, error) {
+	npKey := getPolicyKey(policy)
+	var np *networkPolicy
 
-	if aclLogging.Deny != "" || aclLogging.Allow != "" {
-		klog.Infof("ACL logging for network policy %s in namespace %s set to deny=%s, allow=%s",
-			policy.Name, policy.Namespace, aclLogging.Deny, aclLogging.Allow)
-	}
+	err := oc.networkPolicies.DoWithLock(npKey, func(npKey string) error {
+		oldNP, found := oc.networkPolicies.Load(npKey)
+		if found {
+			cleanupErr := oc.cleanupNetworkPolicy(oldNP)
+			return fmt.Errorf("cleanup for retrying nertwork policy create failed: %v", cleanupErr)
+		}
+		np, found = oc.networkPolicies.LoadOrStore(npKey, NewNetworkPolicy(policy))
+		if found {
+			// that should never happen, because successful cleanup will delete np from oc.networkPolicies
+			return fmt.Errorf("network policy is found in the system, while it should've been cleaned up")
```

Comment:
print the name of the network policy and maybe some relevant info that would help us debug this further